pkg/nfd-worker/nfd-worker.go

```diff
@@ -341,6 +348,12 @@ func (w *nfdWorker) Run() error {
 				return err
 			}
 
+		case <-w.sourceEvent:
+			err = w.runFeatureDiscovery()
```

Comment:
We don't want to run full feature discovery of all sources. Just the one that evented and then update the NodeFeature object. Prolly requires a bit of refactoring on the nfd-worker.go side.

Either do s.Discover() here or then in the source (and then just notify nfd-worker that it needs to re-advertise features).